### PR TITLE
Implement synchronized storage session seam

### PR DESCRIPTION
**Pull request by timbussmann**

### CODE REVIEW

File: src/NServiceBus.RavenDB.AcceptanceTests/ConfigureEndpointRavenDBPersistence.cs [hunk 1]
```diff
@@ -51,11 +51,7 @@ public static DocumentStore GetDocumentStore()
 
     internal static DocumentStore GetInitializedDocumentStore(string defaultDatabase)
     {
-        var urls = Environment.GetEnvironmentVariable("CommaSeparatedRavenClusterUrls");
-        if (urls == null)
-        {
-            throw new Exception("RavenDB cluster URLs must be specified in an environment variable named CommaSeparatedRavenClusterUrls.");
-        }
+        var urls = Environment.GetEnvironmentVariable("CommaSeparatedRavenClusterUrls") ?? "http://localhost:8080";
```
Comment (andreasohlund) [inline src/NServiceBus.RavenDB.AcceptanceTests/ConfigureEndpointRavenDBPersistence.cs:54]: This needs to be rolled back

Comment (andreasohlund) [reply src/NServiceBus.RavenDB.AcceptanceTests/ConfigureEndpointRavenDBPersistence.cs:54]: my bad

Comment (timbussmann) [reply src/NServiceBus.RavenDB.AcceptanceTests/ConfigureEndpointRavenDBPersistence.cs]: I extracted it into a dedicated PR anyway: https://github.com/Particular/NServiceBus.RavenDB/pull/488

File: src/NServiceBus.RavenDB/SessionManagement/ContextBagExtensions.cs [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-    static class ContextBagExtensions
-    {
-        /// <summary>
-        /// Retrieves an IAsyncDocumentSession from the ContextBag. If a sessionFunction exists, that will be used
-        /// to create the session. Otherwise, retrieve the session directly from the bag.
-        /// </summary>
-        internal static IAsyncDocumentSession GetAsyncSession(this ContextBag contextBag)
-        {
-            Func<IAsyncDocumentSession> sessionFunction;
-            contextBag.TryGet(out sessionFunction);
```
Comment (timbussmann) [inline src/NServiceBus.RavenDB/SessionManagement/ContextBagExtensions.cs:16]: I'm not sure what scenario/configuration could ever hit this code path?

Comment (andreasohlund) [reply src/NServiceBus.RavenDB/SessionManagement/ContextBagExtensions.cs:16]: It's internal so removing should be fine?

Comment (timbussmann) [reply src/NServiceBus.RavenDB/SessionManagement/ContextBagExtensions.cs:16]: yeah but I couldn't find a code path which would ever set such a function?

**Pull request merged**